### DISCUSSION TITLE
WireGuard as a VPN client

### DISCUSSION THREAD

**Discussion by YanekJ** [hassio-addons/addon-wireguard#95] [Q&A] [answered]
Hi,
As probably for many users my remote HA host is connected to the network via LTE modem and is behind some CGNATs - meaning DuckDns running on HA itself is not doing any good - can't access the HA from external network, period. I know I could ask my ISP to get a static IP or a VPN - but this is additional cost. 
In my case I already have another ISP delivering internet to my home - and with this ISP the DuckDns works just fine. So, I was thinking: why not to run a VPN server (with port forwarding enabled) in my home network visible as xyz.duckdns.org, then run a VPN client on the HA host? In theory I could then access my HA by directing the traffic to xyz.duckdns.org:8123 which would get forwarded over the encrypted channel to the HA itself... 
Do you see any flaws with this idea? Can WireGuard addon run as a VPN client? 

Thanks,
Yanek

**Comment by frenck** [comment, marked as answer]
No it cannot.

**Comment by YanekJ** [comment]
Thanks for quick answer, I understand the current version may not support that mode. 
Do you think it is technically possible though?

**Reply by frenck** [reply]
Sure, everything is possible. But I would strongly recommend against it on a supervised-enabled machine.
It will mess with the host network and routing, which should not be done by add-ons.

If you need this level of things, you are running the wrong installation type to being with.

**Reply by YanekJ** [reply]
Thanks, just trying to understand (and learn) - running Client VPN on supervised-enabled machine is not recommended, but Server VPN is acceptable? I'm looking for a solution for 'HA behind LTE' problem thus my questions.

**Reply by frenck** [reply]
If you want to solve that problem on a single device, this might not be the right installation method for you.

**Reply by YanekJ** [reply]
Ok, got it. I thought my problem was not that isolated judging by the number of discussions in HA forums, but it may be. Thanks!
